### DISCUSSION TITLE
Static validation of claim in CI pipeline

### DISCUSSION THREAD

**Discussion by chlunde** [crossplane/crossplane#2598] [Q&A] [answered]
Have anyone used https://github.com/yannh/kubeconform or a similar tool to validate claim against the schema in a CI pipeline? Or is this something kyverno can do? It would be nice to give feedback *before* attempting to apply in a GitOps workflow. poking @vfarcic :)

**Comment by eyarz** [comment, marked as answer]
Hi @chlunde, I have tried `kubeconfrom`, `kubeval` and `kubectl dry-run` to validate K8s schemas and I also wrote a post about it ([A Deep Dive Into Kubernetes Schema Validation](https://dev.to/datreeio/a-deep-dive-into-kubernetes-schema-validation-39ll)).

**TL;DR** 
You MUST validate schemas as part of your GitOps workflow and `kubeconfrom` is still the best tool for doing that.
